### PR TITLE
Remove J9 from omrsig* code [Part 2]

### PR DESCRIPTION
**Pull request by babsingh**
- J9 was removed or renamed to OMR in the structures, functions and comments used in the omrsig library. 
- At some places, redundant code was removed.
- A macro with the old name is introduced to prevent breakage in the downstream projects. Such macros can be removed once the old names are no longer used in the downstream projects.
- These changes don't impact external projects.
- Related to https://github.com/eclipse/omr/issues/3713.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by babsingh** [comment]
@youngar can you please review these changes? the renaming is mostly done for internal structures. only the `J9ZOSLEConditionHandlerRecord` structure was noted to be used in downstream projects. for this structure, a macro with the old name is introduced to prevent breakage in downstream projects. I have verified that downstream projects such as OpenJ9 aren't impacted.

fyi - @DanHeidinga @charliegracie

**Comment by babsingh** [comment]
Updated copyrights.

**Comment by babsingh** [comment]
@charliegracie Andrew is on vacation for 2 weeks. Can you please review and merge these changes?

**Comment by charliegracie** [comment]
@genie-omr build all

File: port/zos390/omrsignal_context_ceehdlr.h [hunk 1]
```diff
@@ -43,12 +43,12 @@ typedef struct {
 	uint32_t portLibrarySignalType;
 	void *handlerAddress;
 	void *handlerAddress2;
-} J9LEConditionInfo;
+} OMRLEConditionInfo;
```
Comment (charliegracie) [inline port/zos390/omrsignal_context_ceehdlr.h]: Does this need a typedef similar to J9ZOSLEConditionHandlerRecord?

File: port/zos390/omr__le_api.h [hunk 2]
```diff
@@ -128,6 +128,6 @@ typedef struct j9_31bit_mch {
 	uint32_t padding6[32];
 
 	U_128 vr[32];
-} j9_31bit_mch;
+} omr_31bit_mch;
```
Comment (charliegracie) [inline port/zos390/omr__le_api.h]: Does this need a typedef similar to J9ZOSLEConditionHandlerRecord?

File: port/zos390/omrsignal_context_ceehdlr.h (hunk 1 above)
Comment (babsingh) [reply port/zos390/omrsignal_context_ceehdlr.h]: This struct is only used in OMR. Internal J9 zOS builds didn't complain about missing `J9LEConditionInfo`.

File: port/zos390/omr__le_api.h (hunk 2 above)
Comment (babsingh) [reply port/zos390/omr__le_api.h]: This struct is only used in OMR. Internal J9 zOS builds didn't complain about missing `j9_31bit_mch`.

File: port/zos390/omrsignal_context_ceehdlr.h (hunk 1 above)
Comment (babsingh) [reply port/zos390/omrsignal_context_ceehdlr.h]: I don't think the **J9** structs especially the **ZOS** ones are used by projects other than J9. If you are worried about non-J9 projects, we can add **defines** for the renamed structs. Or we can take some risk and delay the introduction of **defines** if we find other projects are impacted by the renaming.

Comment (babsingh) [reply port/zos390/omrsignal_context_ceehdlr.h]: Also, there is no public function which exposes `OMRLEConditionInfo` and `omr_31bit_mch` to an OMR consumer so it should be safe to exclude the `defines` during renaming. The same applies to `J9ZOSLEConditionHandlerRecord`, which is an internal OMR structure and not exposed via OMR public functions. OpenJ9 shouldn't use `J9ZOSLEConditionHandlerRecord` directly. Do we guarantee support for private structures used within OMR? If not, we can remove the `define` for `J9ZOSLEConditionHandlerRecord` from this pull request, and break J9 and force it to amend their implementation.

**Comment by charliegracie** [comment]
The Travis failure is an infrastructure failure.  I tried restarting and it did not work. Merging as-is

**Pull request merged**